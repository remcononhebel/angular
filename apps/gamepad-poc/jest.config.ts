/* eslint-disable */
export default {
  displayName: 'gamepad-poc',
  preset: '../../jest.preset.js',
  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  globals: {},
  coverageDirectory: '../../coverage/apps/gamepad-poc',
  coveragePathIgnorePatterns: ['.jest-mock.ts'],
  transform: {
    '^.+\\.(ts|mjs|js|html)$': [
      'jest-preset-angular',
      {
        tsconfig: '<rootDir>/tsconfig.spec.json',
        stringifyContentPathRegex: '\\.(html|svg)$',
      },
    ],
  },
  transformIgnorePatterns: ['node_modules/(?!.*\\.mjs$)'],
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment',
  ],
};
