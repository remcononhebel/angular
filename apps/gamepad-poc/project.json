{
  "name": "gamepad-poc",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "projectType": "application",
  "prefix": "app",
  "sourceRoot": "apps/gamepad-poc/src",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@angular-devkit/build-angular:application",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/apps/gamepad-poc",
        "index": "apps/gamepad-poc/src/index.html",
        "browser": "apps/gamepad-poc/src/main.ts",
        "polyfills": ["zone.js"],
        "tsConfig": "apps/gamepad-poc/tsconfig.app.json",
        "inlineStyleLanguage": "scss",
        "assets": [
          "apps/gamepad-poc/src/favicon.ico",
          "apps/gamepad-poc/src/assets"
        ],
        "styles": ["apps/gamepad-poc/src/styles.scss"],
        "scripts": []
      },
      "configurations": {
        "production": {
          "budgets": [
            {
              "type": "initial",
              "maximumWarning": "500kb",
              "maximumError": "1mb"
            },
            {
              "type": "anyComponentStyle",
              "maximumWarning": "2kb",
              "maximumError": "4kb"
            }
          ],
          "outputHashing": "all"
        },
        "development": {
          "optimization": false,
          "extractLicenses": false,
          "sourceMap": true
        }
      },
      "defaultConfiguration": "production"
    },
    "serve": {
      "executor": "@angular-devkit/build-angular:dev-server",
      "configurations": {
        "production": {
          "buildTarget": "gamepad-poc:build:production"
        },
        "development": {
          "buildTarget": "gamepad-poc:build:development"
        }
      },
      "defaultConfiguration": "development"
    },
    "extract-i18n": {
      "executor": "@angular-devkit/build-angular:extract-i18n",
      "options": {
        "buildTarget": "gamepad-poc:build"
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "apps/gamepad-poc/jest.config.ts"
      }
    },
    "serve-static": {
      "executor": "@nx/web:file-server",
      "options": {
        "buildTarget": "gamepad-poc:build",
        "staticFilePath": "dist/apps/gamepad-poc/browser"
      }
    }
  }
}
