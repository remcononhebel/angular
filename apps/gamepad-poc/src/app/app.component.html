<h1>Hello, World!</h1>

<div>
  <app-gamepad-poc></app-gamepad-poc>
</div>
