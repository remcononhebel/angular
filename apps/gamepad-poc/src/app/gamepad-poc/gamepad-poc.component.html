<div class="debug">
  <div class="debug__joystick-info">
    <h2>Left joystick</h2>
    <pre>{{ left$ | async | json }}</pre>
  </div>

  <div class="debug__joystick-info">
    <h2>Right joystick</h2>
    <pre>{{ right$ | async | json }}</pre>
  </div>
</div>

<div class="joysticks">
  <app-joystick
    [joystickId]="'leftJoystick'"
    [options]="leftJoystickOptions"
    (move)="onLeftJoystickMove($event)"></app-joystick>

  <app-joystick
    [joystickId]="'rightJoystick'"
    [options]="rightJoystickManagerOptions"
    (move)="onRightJoystickMove($event)"></app-joystick>
</div>
