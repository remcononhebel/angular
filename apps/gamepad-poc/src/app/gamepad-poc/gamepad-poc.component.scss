.debug {
  display: flex;
  justify-content: space-between;

  &__joystick-info {
    background-color: #ddd;
    border: 1px solid black;
    padding: 8px;
    width: 50%;
  }
}

.joysticks {
  position: absolute;
  bottom: 0;
  width: 100%;
  display: flex;
  justify-content: space-between;
}
