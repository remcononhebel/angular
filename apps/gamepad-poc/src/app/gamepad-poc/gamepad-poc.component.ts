import { CommonModule } from '@angular/common';
import { Component } from '@angular/core';
import nipplejs from 'nipplejs';
import { Subject, map } from 'rxjs';
import { JoystickComponent } from '../joystick/joystick.component';

export interface JoystickViewModel {
  direction: nipplejs.Direction;
}

@Component({
  selector: 'app-gamepad-poc',
  standalone: true,
  imports: [CommonModule, JoystickComponent],
  templateUrl: './gamepad-poc.component.html',
  styleUrl: './gamepad-poc.component.scss',
})
export class GamepadPocComponent {
  readonly leftJoystickOutputData$ = new Subject<nipplejs.JoystickOutputData>();
  readonly rightJoystickOutputData$ = new Subject<nipplejs.JoystickOutputData>();

  readonly left$ = this.leftJoystickOutputData$.pipe(map((data) => this.mapToViewModel(data)));
  readonly right$ = this.rightJoystickOutputData$.pipe(map((data) => this.mapToViewModel(data)));

  leftJoystickManager?: nipplejs.JoystickManager;
  rightJoystickManager?: nipplejs.JoystickManager;

  readonly leftJoystickOptions: nipplejs.JoystickManagerOptions = {
    mode: 'static',
    position: { left: '100px', bottom: '100px' },
    color: 'red',
  };

  readonly rightJoystickManagerOptions: nipplejs.JoystickManagerOptions = {
    mode: 'static',
    position: { left: '100px', bottom: '100px' },
    color: 'blue',
  };

  onLeftJoystickMove(event: nipplejs.JoystickOutputData) {
    this.leftJoystickOutputData$.next(event);
  }

  onRightJoystickMove(event: nipplejs.JoystickOutputData) {
    this.rightJoystickOutputData$.next(event);
  }

  private mapToViewModel(data: nipplejs.JoystickOutputData): JoystickViewModel | null {
    return { direction: data.direction };
  }
}
