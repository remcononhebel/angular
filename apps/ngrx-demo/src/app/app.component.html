<div class="app">
  <div class="app__available-books">
    <h1>Available books ({{ numberOfAvailableBooks$ | async }})</h1>
    <ngrx-demo-book-list
      class="book-list"
      [books]="(books$ | async)!"
      (add)="onAdd($event)"></ngrx-demo-book-list>
    <div class="app__action">
      <button
        (click)="selectAll()">
        Add all to collection
      </button>
    </div>
  </div>

  <div class="app__my-collection">
    <h1>My Collection ({{ numberOfBooksInCollections$ | async }})</h1>
    <ngrx-demo-book-collection
      class="book-collection"
      [books]="(bookCollection$ | async)!"
      (remove)="onRemove($event)">
    </ngrx-demo-book-collection>
    <div class="app__action">
      <button
        (click)="deselectAll()">
        Remove all from collection
      </button>
    </div>
  </div>
</div>
