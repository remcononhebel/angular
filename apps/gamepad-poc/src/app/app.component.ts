import { Component } from '@angular/core';
import { JoystickPocComponent } from './gamepad-poc/gamepad-poc.component';

export interface JoystickViewModel {}

@Component({
  standalone: true,
  imports: [JoystickPocComponent],
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrl: './app.component.scss',
})
export class AppComponent {}
