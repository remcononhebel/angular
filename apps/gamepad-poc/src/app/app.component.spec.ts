import { ComponentFixture } from '@angular/core/testing';
import { Spectator, createComponentFactory } from '@ngneat/spectator';
import { MockComponent } from 'ng-mocks';
import { AppComponent } from './app.component';
import { GamepadPocComponent } from './gamepad-poc/gamepad-poc.component';

describe('AppComponent', () => {
  let spectator: Spectator<AppComponent>;
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  const createComponent = createComponentFactory({
    component: AppComponent,
    shallow: true,
    imports: [MockComponent(GamepadPocComponent)],
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
    fixture = spectator.fixture;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render a header and the app-gamepad-poc component', () => {
    expect(fixture).toMatchSnapshot();
  });
});
