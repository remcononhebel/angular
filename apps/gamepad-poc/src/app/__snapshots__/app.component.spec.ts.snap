// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render a header and the app-gamepad-poc component 1`] = `
<app-root>
  <h1>
    Hello, World!
  </h1><div>
    <app-gamepad-poc />
  </div>
</app-root>
`;
